priv/www/index.html

```diff
@@ -16,43 +17,54 @@
     <script src="js/prefs.js" type="text/javascript"></script>
     <script src="js/formatters.js" type="text/javascript"></script>
     <script src="js/charts.js" type="text/javascript"></script>
-    <script src="js/singular/singular.js" type="application/javascript"></script>
-    
+    <script src="js/singular/singular.js" type="application/javascript"></script>    
     <link href="css/main.css" rel="stylesheet" type="text/css"/>
     <link href="favicon.ico" rel="shortcut icon" type="image/x-icon"/>
     
     <script type="application/javascript">
-      var uaa_logged_in = false;
+      var oauth2_user_logged_in = false;
       var uaa_invalid = false;
       var auth = JSON.parse(sync_get('/auth'));
       enable_uaa = auth.enable_uaa;
+      oauth2_implementation = auth.oauth2_implementation;
       uaa_client_id = auth.uaa_client_id;
       uaa_location = auth.uaa_location;
+      oauth2_scopes = auth.oauth2_scopes;
+      oauth2_implementation = oauth2_implementation.toLowerCase()
+      
       if (enable_uaa) {
-        Singular.init({
-         singularLocation: './js/singular/',
-         uaaLocation: uaa_location,
-         clientId: uaa_client_id,
-         onIdentityChange: function (identity) {
-          uaa_logged_in = true;
-          start_app_login();
-         },
-         onLogout: function () {
-          uaa_logged_in = false;
-          var hash = window.location.hash.substring(1);
-          var params = {}
-          hash.split('&').map(hk => {
-           let temp = hk.split('=');
-           params[temp[0]] = temp[1]
-          });
-          if (params.error) {
-           uaa_invalid = true;
-           replace_content('login-status', '<p class="warning">' + decodeURIComponent(params.error) + ':' + decodeURIComponent(params.error_description) + '</p> <button id="loginWindow" onclick="uaa_login_window()">Click here to log out</button>');
-          } else {
-           replace_content('login-status', '<button id="loginWindow" onclick="uaa_login_window()">Click here to log in</button>');
-          }
-         }
-      });
+        switch (oauth2_implementation) {
```

Comment:
Perhaps we should extract two different functions here.

Reply:
Would you like to encapsulate each case to separate functions, right? Theirs names like initialize uaa() / initialize identityserver() would it be ok?

Reply:
Yes, those names sound good to me.

Reply:
Done